broadinstitute/single_cell_portal_core
# adding GOOGLE_PROJECT_NUMBER to cron_env

#### BACKGROUND & CHANGES
This update adds the environment variable `GOOGLE_PROJECT_NUMBER` to the cron env file inside the portal Docker container.  This value is necessary for launching ingest jobs as it is a path-level URL parameter for job submission.  While very rare, if the DelayedJob manager ever detects that workers have died, it will manually restart all job workers, using the cron env file to source all necessary secrets.  This causes the project number to be unset, and then all future ingest jobs will fail until the portal container is completely restarted.

#### MANUAL TESTING
Testing this locally is cumbersome, as `rails_startup.bash` is only ever invoked in deployed environments.  Furthermore, this script is added to compiled Docker image via our CI/CD pipeline, meaning you will have to manually rebuild the image locally to test.  To do so:
1. Pull this branch first
2. Load all secrets into your environment with `./rails_local_setup.rb && source config/secrets/.source_env.bash`
3. Rebuild the development image:
```
IMAGE_NAME='gcr.io/broad-singlecellportal-staging/single-cell-portal:development'
docker build -t $IMAGE_NAME .
```
4. Once that completes (it will take several minutes), run `bin/boot_docker` 
5. Wait until you see the following lines output in the terminal:
```
*** CREATING CRON ENV FILES ***
*** COMPLETED ***
``` 
6. In a separate terminal window, attach to the running container with `docker exec -it single_cell bash` and then run the following command to verify that `GOOGLE_PROJECT_NUMBER` has been set:
```
cat /home/app/.cron_env | grep GOOGLE_PROJECT_NUMBER
=> export GOOGLE_PROJECT_NUMBER='[your project number]'
```
